From 204fcceb7ccf43034da8e97078153c7c6d0bc84d Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Wed, 28 Apr 2021 13:17:47 +0900
Subject: [PATCH] cifsd: add ksmbd/nfsd interoperability to feature table
Git-commit: 204fcceb7ccf43034da8e97078153c7c6d0bc84d
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


Add ksmbd/nfsd interoperability to feature table and sync with a table in
patch cover letter.

Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 Documentation/filesystems/cifs/cifsd.rst | 32 ++++++++++++++++--------
 1 file changed, 22 insertions(+), 10 deletions(-)

diff --git a/Documentation/filesystems/cifs/cifsd.rst b/Documentation/filesystems/cifs/cifsd.rst
index cb9f87b8529f..01a0be272ce6 100644
--- a/Documentation/filesystems/cifs/cifsd.rst
+++ b/Documentation/filesystems/cifs/cifsd.rst
@@ -67,7 +67,8 @@ CIFSD Feature Status
 Feature name                   Status
 ============================== =================================================
 Dialects                       Supported. SMB2.1 SMB3.0, SMB3.1.1 dialects
-                               excluding security vulnerable SMB1.
+                               (intentionally excludes security vulnerable SMB1
+                               dialect).
 Auto Negotiation               Supported.
 Compound Request               Supported.
 Oplock Cache Mechanism         Supported.
@@ -79,26 +80,37 @@ HMAC-SHA256 Signing            Supported.
 Secure negotiate               Supported.
 Signing Update                 Supported.
 Pre-authentication integrity   Supported.
-SMB3 encryption(CCM, GCM)      Supported.
-SMB direct(RDMA)               Partial Supported. SMB3 Multi-channel is required
-                               to connect to Windows client.
+SMB3 encryption(CCM, GCM)      Supported. (CCM and GCM128 supported, GCM256 in
+                               progress)
+SMB direct(RDMA)               Partially Supported. SMB3 Multi-channel is
+                               required to connect to Windows client.
 SMB3 Multi-channel             In Progress.
 SMB3.1.1 POSIX extension       Supported.
-ACLs                           Partial Supported. only DACLs available, SACLs is
-                               planned for future. ksmbd generate random subauth
+ACLs                           Partially Supported. only DACLs available, SACLs
+                               (auditing) is planned for the future. For
+                               ownership (SIDs) ksmbd generates random subauth
                                values(then store it to disk) and use uid/gid
                                get from inode as RID for local domain SID.
                                The current acl implementation is limited to
                                standalone server, not a domain member.
+                               Integration with Samba tools is being worked on
+                               to allow future support for running as a domain
+                               member.
 Kerberos                       Supported.
 Durable handle v1,v2           Planned for future.
 Persistent handle              Planned for future.
 SMB2 notify                    Planned for future.
 Sparse file support            Supported.
-DCE/RPC support                Partial Supported. a few calls(NetShareEnumAll,
-                               NetServerGetInfo, SAMR, LSARPC) that needed as
-                               file server via netlink interface from
-                               ksmbd.mountd.
+DCE/RPC support                Partially Supported. a few calls(NetShareEnumAll,
+                               NetServerGetInfo, SAMR, LSARPC) that are needed
+                               for file server handled via netlink interface
+                               from ksmbd.mountd. Additional integration with
+                               Samba tools and libraries via upcall is being
+                               investigated to allow support for additional
+                               DCE/RPC management calls (and future support
+                               for Witness protocol e.g.)
+ksmbd/nfsd interoperability    Planned for future. The features that ksmbd
+                               support are Leases, Notify, ACLs and Share modes.
 ============================== =================================================
 
 
-- 
2.33.0

